owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,326 @@
+        val newDirectory = "newDirectory"
+
+        ocLocalTransferDataSource.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+        }
+    }
+
+    @Test
+    fun `deleteTransferById removes it correctly`() {
+
+        ocLocalTransferDataSource.deleteTransferById(id)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun `deleteAllTransfersFromAccount removes all transfers correctly`() {
+
+        ocLocalTransferDataSource.deleteAllTransfersFromAccount(OC_ACCOUNT_NAME)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        }
+    }
+
+    @Test
+    fun `getTransferById returns a OCTransfer`() {
+        every {
+            transferDao.getTransferWithId(any())
+        } returns ocTransfer.toEntity()
+
+        val actualResult = ocLocalTransferDataSource.getTransferById(id)
+
+        assertEquals(ocTransfer, actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun `getAllTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getAllTransfers()
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getAllTransfers()
+        }
+    }
+
+    @Test
+    fun `getAllTransfersAsStream returns a flow of list of OCTransfer ordered by status`() = runBlocking {
+
+        val transferEntityInProgress: OCTransferEntity = ocTransfer.toEntity().copy(status = 0)
+
+        val transferEntityQueue: OCTransferEntity = ocTransfer.toEntity().copy(status = 1)
+
+        val transferEntityFailed: OCTransferEntity = ocTransfer.toEntity().copy(status = 2)
+
+        val transferEntitySucceeded: OCTransferEntity = ocTransfer.toEntity().copy(status = 3)
+
+        val transferListRandom = listOf(transferEntityQueue, transferEntityFailed, transferEntityInProgress, transferEntitySucceeded)
+
+
+        val transferQueue = ocTransfer.copy()
+        transferQueue.status =  TransferStatus.TRANSFER_QUEUED
+
+        val transferFailed = ocTransfer.copy()
+        transferFailed.status =  TransferStatus.TRANSFER_FAILED
+
+        val transferSucceeded = ocTransfer.copy()
+        transferSucceeded.status =  TransferStatus.TRANSFER_SUCCEEDED
+
+        val transferListOrdered = listOf(ocTransfer, transferQueue, transferFailed, transferSucceeded)
+
+        every {
+            transferDao.getAllTransfersAsStream()
+        } returns flowOf(transferListRandom)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfersAsStream().first().map { it }
```

Comment:
Again, does this `map { it }` make sense?

Reply:
Done